### ISSUE TITLE
What analytics do we want to track?

### ISSUE THREAD

**Issue by phargogh** [natcap/data.naturalcapitalproject.stanford.edu#33]
The topic of analytics has come up regularly, but it would be very helpful if we could, before tossing an analytics solution on the site and calling it good, decide what questions we want to answer with analytics and then design the analytics solution around the questions.

In case we need it, a google analytics extension is available at https://github.com/ckan/ckanext-googleanalytics.  I have, for a minute there, been able to get this working on the staging site.

**Comment by empavia** [comment]
Thanks for linking us all to this issue, James. Here are my suggestions:

- Which datasets are downloaded the most?
- Which InVEST models are searched the most? (ex. How often is each model from the dropdown selected?)
- What are the most common filters applied (both at the top-level, such as use of Keyword search, and the specific-level, such as specific keywords)? 
- In which geographic areas is the hub accessed from?
- In which geographic areas are data downloaded?
- How many website visits are we getting, in general (maybe just on the landing page)?
-How many people download full dataset vs. use the clipping app?

Curious what you all think! These are the main items that stick out to me.

**Comment by lmandle** [comment]
It's great to be thinking through and discussing this, @phargogh and @empavia!
 
I've been thinking about wanting to be able to answer the following questions to support decision-making:
- How much is the Hub being used overall and who is using it? (To help us raise ongoing funding for NatCap/the Data Hub and inform how resources are allocated to the Data Hub; is it a resource mostly for NatCappers? People from particular countries?)
- How often are different datasets searched for and downloaded? (Are there datasets that are rarely used and so we might want to drop/not need to update? Are there popular datasets that are a priority for updating/improving? Are there datasets that are commonly searched for but not downloaded, and could we figure out what the gap is?)
- What are people searching for through the search bar or through filters/keywords? (Are there things people are looking for that we are not providing and should be, or could provide more of? Is there demand more for InVEST input data or results or something else?)

So, in line with many of the suggestions @empavia already made, that leads me to wanting to track things like:
- Number of visitors to the Data Hub landing page (and any sort of information on them - geography, etc.) + what number and proportion of visitors search for data? + What number and proportion end up downloading data?
- Number of downloads over time for the different data sets + info on who is doing the downloading and where, if available
- Search terms being entered

**Comment by phargogh** [comment]
So I guess I hadn't enabled `tracking` in the `ckan.plugins` list, and when I did this on the production instance, I inadvertently caused a second search bar to show up!  So there's clearly some kind of templating funkiness going on here that I'll need to resolve.

<img width="1905" alt="Image" src="https://github.com/user-attachments/assets/7d32e07e-00f0-42d1-81f3-7e15a46c433f" />

**Comment by phargogh** [comment]
# Plugin order really matters with a system of template overrides!

in `.env`, I need to have `natcap` listed before (to the left of) `tracking` in order for our theme updates to override the updates that `ckanext-tracking` defines.

Added to the example plugins in 4314521

**Comment by phargogh** [comment]
Whoops, reopening because I haven't solved the other tracking issues.

**Comment by phargogh** [comment]
Google analytics is now enabled on both staging and prod, reporting to our same GA4 account as of 5e57b700e8b31116027dde765f093409d81c538d

**Comment by phargogh** [comment]
Now that we have all 3 forms of tracking enabled, we need to verify that we can answer the questions stated.

* Questions related to raw data accesses (including those not made through ckan) are tracked through bucket logging.  Some of these will be double-counted with CKAN-land resource downloads.
* Questions related to CKAN page visits and resource downloads are tracked by internal CKAN tracking.
* Questions related to URLs, including page visits, resource downloads (I think) and search terms (I think) should be handled by  Google Analytics.

Specific unknowns that need to be verified include:
 - [ ] can we access the specific search parameters used in GA logs?
 - [ ] are InVEST models reflected in GA tracking?
 - [ ] can we identify filters applied in GA tracking?

I should be able to test this in the GA dashboard in the next couple of days.

**Comment by phargogh** [comment]
> Can we access the specific search parameters used in GA logs?

Yes, example: `/dataset/?q=Auckland&sort=score+desc%2C+metadata_modified+desc&ext_bbox=`

Once we start collecting a whole lot of traffic, it will be interesting to see how many of these unique sets of search parameters will remain in the GA logs before some are sampled out.  They are tracked, in any case, and we will be able to at least see some patterns, like if there's a _really_ common search query, I'd expect that to reliably show up in the GA logs.

> Are InVEST models reflected in GA tracking?

Yes.  Example: `/dataset/?invest_model=Nutrient+Delivery+Ratio`

> can we identify filters applied in GA tracking?

Yes, these show up in the URL as well, such as `/dataset/?extras_sources_res_formats=shp`

**Comment by phargogh** [comment]
Going back through @lmandle and @empavia's comments, here are things that we haven't yet resolved with tracking:

* How many people are using the clipping app?
   * This is not yet explicitly tracked, but I also think this will be better tracked from within the server-side app than from google analytics.
* How many users of the data hub are natcappers?
   * This is something we won't be able to resolve completely without asking natcappers to log in, but I suspect we'll probably be able to have a decent guess at this from the locations of downloads.
* ~~How often are the individual files downloaded through the data catalog interface?~~
   * Actually, this is accessible in google analytics using the `Link URL` dimension in a data exploration window.

**Comment by phargogh** [comment]
I have created two new issues for the remaining analytics tasks: https://github.com/natcap/data.naturalcapitalproject.stanford.edu/issues/84 and https://github.com/natcap/data.naturalcapitalproject.stanford.edu/issues/85
